### PR TITLE
Only build docs on ExDoc main on CI

### PR DESCRIPTION
**Pull request by josevalim**
The new repo.hex.pm action uses latest ExDoc already.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,21 +50,15 @@ jobs:
...
-            cd ..
-            git clone https://github.com/elixir-lang/ex_doc.git --branch ${branch} --depth 1
-            cd ex_doc
-            ../elixir/bin/mix do local.rebar --force + local.hex --force + deps.get + compile
-            cd ../elixir/
-            make docs
-            rm -rf ../ex_doc/
-          done
+          cd ..
+          git clone https://github.com/elixir-lang/ex_doc.git --branch ${branch} --depth 1
```
Comment (josevalim) [inline .github/workflows/ci.yml:57]:
```suggestion
          git clone https://github.com/elixir-lang/ex_doc.git --depth 1
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -50,21 +50,15 @@ jobs:
...
-            cd ex_doc
-            ../elixir/bin/mix do local.rebar --force + local.hex --force + deps.get + compile
-            cd ../elixir/
-            make docs
-            rm -rf ../ex_doc/
-          done
+          cd ..
+          git clone https://github.com/elixir-lang/ex_doc.git --depth 1
+          cd ex_doc
+          ../elixir/bin/mix do local.rebar --force + local.hex --force + deps.get + compile
```
Comment (ericmj) [inline .github/workflows/ci.yml:59]: The `+` syntax is new to me 😲, much better than commas.

**Review by ericmj — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (josevalim) [reply .github/workflows/ci.yml:59]: The comma one didn't work on Windows under some scenarios, so we had to pick something else.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**